oxia-operator/controllers/oxiacluster_resources.go

```diff
@@ -0,0 +1,339 @@
+// Copyright 2023 StreamNative, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controllers
+
+import (
+	"fmt"
+	monitoringV1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	oxiav1alpha1 "github.com/streamnative/oxia/api/v1alpha1"
+	"golang.org/x/exp/maps"
+	"gopkg.in/yaml.v2"
+	appsV1 "k8s.io/api/apps/v1"
+	coreV1 "k8s.io/api/core/v1"
+	rbacV1 "k8s.io/api/rbac/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+)
+
+func resourceName(component Component, name string) string {
+	if component == Coordinator {
+		return name + "-coordinator"
+	}
+	return name
+}
+
+func objectMeta(component Component, oxia *oxiav1alpha1.OxiaCluster) metaV1.ObjectMeta {
+	_resourceName := resourceName(component, oxia.Name)
+	return metaV1.ObjectMeta{
+		Name:      _resourceName,
+		Namespace: oxia.Namespace,
+		Labels:    allLabels(component, oxia.Name),
```

Comment:
Consider add `OwnerReferences`, which will ensure that resources are automatically cleaned up